src/ConstructiveSolidGeometry/SurfacePrimitives/RegularPolygon.jl

```diff
@@ -71,10 +98,29 @@ function sample(rp::RegularPolygon{N,T}, Nsamps::NTuple{3,Int} = (2,N+1,2))::Vec
         ]
 end
 
+
 function sample(rp::RegularPolygon{N,T}, g::CylindricalTicksTuple{T})::Vector{CylindricalPoint{T}} where {N,T}
     samples = [
         CylindricalPoint{T}(r,φ,rp.z)
         for φ in _get_ticks(g.φ, T(0), T(2π))
         for r in _get_ticks(g.r, get_r_at_φ(rp,φ)...)
     ]
 end
+
+function sample(rp::RegularPolygon{N,T}, g::CartesianTicksTuple{T})::Vector{CartesianPoint{T}} where {N,T}
+    corners = SVector{N+1,Tuple{T,T}}(sincos.(T(2π/N) .* (0:N)))
+    rMin::T, rMax::T = get_r_limits(rp)
+    samples = vcat(
```

Comment:
Will `vcat` lead to poor performance (and if so, how can we avoid it)?
I checked: sampling a `RegularPolygon` or `RegularPrismMantle` takes approximately as long as sampling a `ConeMantle`

Reply:
From my checks, it looks like it's the same speed as `append!`